src/AppInstallerRepositoryCore/Microsoft/PredefinedInstalledSourceFactory.cpp

```diff
@@ -68,14 +68,50 @@ namespace AppInstaller::Repository::Microsoft
 
             IIterable<Package> packages;
             PackageManager packageManager;
+
             if (scope == Manifest::ScopeEnum::Machine)
             {
-                packages = packageManager.FindProvisionedPackages();
+                // May not be present on our oldest supported systems; simply ignore for the time being.
+                IPackageManager9 packageManager9 = packageManager.try_as<IPackageManager9>();
+                if (packageManager9)
+                {
+                    packages = packageManager.FindProvisionedPackages();
+                }
+                else
+                {
+                    AICLI_LOG(Repo, Warning, << "FindProvisionedPackages is not available on this version of Windows");
+                }
             }
             else
             {
                 // TODO: Consider if Optional packages should also be enumerated
-                packages = packageManager.FindPackagesForUserWithPackageTypes({}, PackageTypes::Main | PackageTypes::Framework);
+                for (PackageTypes types : { PackageTypes::Main | PackageTypes::Framework, PackageTypes::Main, PackageTypes::Framework })
+                {
+                    try
+                    {
+                        packages = packageManager.FindPackagesForUserWithPackageTypes({}, types);

```

Comment:
if Main | Framework failed, does enumerating Main and Framework individually and combining the results work?

Reply:
My assumption is that this issue is caused by some package's state being corrupted.  Hopefully if it is just one package, then it must either be a Main OR a Framework.  Attempting to get that one will also fail, and we can enumerate the other list.  But this is all speculation on the actual issue.

A MUCH larger change would surface this inability to enumerate the packages and allow the code with context to decide how to proceed like we do with the errors from available catalogs.  But on thinking through the implications here, I think it is better to allow reduced functionality rather than hard error.  In the cases where it matters, it probably means doing extra work.  In the cases where it doesn't matter, it means success instead of the current error.